src/Http/Http.Abstractions/src/Validation/ValidateContext.cs

```diff
@@ -59,42 +60,139 @@ public sealed class ValidateContext
     /// This is used to prevent stack overflows from circular references.
     /// </summary>
     public int CurrentDepth { get; set; }
+    
+    /// <summary>
+    /// Gets or sets the JSON serializer options to use for property name formatting.
+    /// When set, property names in validation errors will be formatted according to the
+    /// PropertyNamingPolicy and JsonPropertyName attributes.
+    /// </summary>
+    internal JsonSerializerOptions? SerializerOptions { get; set; }
```

Comment:
Still not quite right. You need to do something like this:

```suggestion
    internal JsonSerializerOptions? SerializerOptions => ValidationContext.GetService(typeof(JsonOptions))?.SerializerOptions;
```

Then, you don't need to set the value in ValidationEndpointFilterFactory.